test/realtime/tenants/connect/piper_test.exs

```diff
@@ -0,0 +1,66 @@
+defmodule Realtime.Tenants.Connect.PiperTest do
+  use ExUnit.Case
```

Comment:
If we can make this asynchronous we definitely should, if not, consider adding an explicit `async: false` with a comment explaining why not. We like to do that sort of thing because looking at such code again down the road gives us an opportunity to check if the status quo have changed since and perhaps we can make it async again.

```suggestion
  use ExUnit.Case, async: true
```